src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -284,40 +310,87 @@ export default class PhotoField extends React.Component {
   }
 
   onZoom = (e) => {
+    // Cropper returns the attempted zoom value
     const zoomValue = e.detail.ratio;
-    if (zoomValue < MAX_RATIO && zoomValue > MIN_RATIO) {
-      this.setState({ zoomValue });
-      this.updateBoundaryWarningAndButtonStates();
 
-      const moveBoundariesMet = getMoveBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
-      const zoomWarn = zoomValue >= WARN_RATIO;
+    // check if zoom is out of bounds
+    let zoomToBoundaryRatio;
+    if (zoomValue < this.state.minRatio) {
+      zoomToBoundaryRatio = this.state.minRatio;
+    } else if (zoomValue > this.state.maxRatio) {
+      zoomToBoundaryRatio = this.state.maxRatio;
+    }
 
-      this.setState({
-        warningMessage: makeCropBoundaryWarningMessage({ zoomWarn, ...moveBoundariesMet }),
-        ...makeMoveButtonsEnabledStates(moveBoundariesMet)
-      });
-    } else {
-      e.preventDefault();
+    // force zoom within zoom bounds
+    if (zoomToBoundaryRatio) {
+      this.refs.cropper.zoomTo(zoomToBoundaryRatio); // force zoom within constraints
+      e.preventDefault(); // prevents bad zoom attempt
+      return; // don't update state until the subsequent zoom attempt
     }
+
+    // zoom value is good- update the state / messaging
+    this.setState({ zoomValue });
+
+    const zoomWarn = zoomValue >= WARN_RATIO;
+    // at this point, the onZoom event has not resized the canvas-
+    // this forces the canvas back into move bounds if the zoom pulls a canvas edge into the cropBox
+    //   after the canvas has rendered with new width values
+    window.requestAnimationFrame(this.maybeMoveCanvasWithinBounds.bind(this, {}, zoomWarn));
```

Comment:
Why did you choose to use `bind` over

```js
window.requestAnimationFrame(() => this.maybeMoveCanvasWithinBounds({}, zoomWarn))
```

I find that slightly easier to read than using `bind`, but don't have a strong opinion about it.

Reply:
Updated. For some reason I thought `this` wouldn't be maintained with the right context without binding it.

Reply:
We're using ES.next syntax for class methods that does the binding:

`someMethod = () => { }`

vs

`someMethod() { }`

I think it would actually work without the latter in this case, too, but this is why you don't see more `bind` calls in our components.